addons/hr_expense/models/hr_expense.py

```diff
@@ -199,11 +200,18 @@ def _compute_amount_residual(self):
                 .filtered(lambda line: line.expense_id == expense and line.account_type in ('asset_receivable', 'liability_payable'))
             expense.amount_residual = -sum(payment_term_lines.mapped(residual_field))
 
-    @api.depends('currency_rate', 'total_amount', 'amount_tax')
+    @api.depends('currency_rate', 'total_amount', 'tax_ids', 'product_id', 'employee_id.user_id.partner_id', 'quantity')
     def _compute_total_amount_company(self):
         for expense in self:
-            expense.total_amount_company = expense.total_amount * expense.currency_rate
-            expense.amount_tax_company = expense.amount_tax * expense.currency_rate
+            taxes = expense.tax_ids.with_context(force_price_include=True, round=True).compute_all(
```

Comment:
It would be better to have `_convert_to_tax_base_line_dict` on expense and use `_compute_taxes` everywhere